src/cpp/session/modules/build/SessionInstallRtools.cpp

```diff
@@ -89,6 +92,31 @@ Error installRtools()
    if (error)
       return error;
 
+   if (version == "4.2")
+   {
+      try
+      {
+         FilePath tmpFile;
+         FilePath::tempFilePath(tmpFile);
+         r::exec::RFunction("utils:::download.file")
+               .addParam("url", url)
+               .addParam("destfile", tmpFile.getAbsolutePath())
+               .addParam("mode", "rb")
+               .call();
+
+         std::string rToolsHome = file_utils::readFile(tmpFile);
+         boost::regex reLinkPattern("<a\\shref=\"(.*\\.exe)\">\\s*Rtools42 installer<\\/a>", boost::regex::icase);
+         boost::cmatch matches;
```

Comment:
You could also use `boost::smatch` and match on the `std::string` directly (rather than the `c_str()`).